File: app/src/main/java/com/shoegazerwithak/lesswrongeveryday/model/Article.java
package com.shoegazerwithak.lesswrongeveryday.model;

import android.os.Parcel;
import android.os.Parcelable;
import android.text.TextUtils;

/**
 * POJO to save data from JSON
 */

public class Article implements Parcelable {

    /**
     * Parcelable stuff
     **/

    public static final Creator<Article> CREATOR = new Creator<Article>() {
        @Override
        public Article createFromParcel(Parcel in) {
            return new Article(in);
        }

        @Override
        public Article[] newArray(int size) {
            return new Article[size];
        }
    };
//    public int id;
    public String title;
    public String link;
    public String text;

    public Article(String mTitle, String mLink) {
        title = mTitle;
        link = mLink;
    }

    public Article(String mTitle, String mLink, String mText) {
        title = mTitle;
        link = mLink;
        text = mText;
    }

    public Article(Parcel in) {
//        id = in.readInt();
        title = in.readString();
        link = in.readString();
        text = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
//        parcel.writeInt(id);
        parcel.writeString(title);
        parcel.writeString(link);
        parcel.writeString(text);
    }
}

File: app/src/main/java/com/shoegazerwithak/lesswrongeveryday/model/Article.kt
package com.shoegazerwithak.lesswrongeveryday.model

import android.os.Parcel
import android.os.Parcelable

/**
 * POJO to save data from JSON
 */

class Article : Parcelable {
    //    public int id;
    var title: String
    var link: String
    var text: String = ""

    constructor(mTitle: String, mLink: String, mText: String = "") {
        title = mTitle
        link = mLink
        text = mText
    }

    constructor(`in`: Parcel) {
        //        id = in.readInt();
        title = `in`.readString()
        link = `in`.readString()
        text = `in`.readString()
    }

    override fun describeContents(): Int = 0

    override fun writeToParcel(parcel: Parcel, i: Int) {
        //        parcel.writeInt(id);
        parcel.writeString(title)
        parcel.writeString(link)
        parcel.writeString(text)
    }

    companion object {
        /**
         * Parcelable stuff
         */
        val CREATOR: Parcelable.Creator<Article> = object : Parcelable.Creator<Article> {
            override fun createFromParcel(`in`: Parcel): Article {
                return Article(`in`)
            }

            override fun newArray(size: Int): Array<Article?> = arrayOfNulls(size)
        }
    }
}